src/Events/Category_Colors/Meta_Keys.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Meta Keys Helper for Category Colors.
+ *
+ * This class manages the meta keys used for storing category color data.
+ * It provides utility methods for retrieving prefixed meta keys and a list of all available keys.
+ *
+ * @since   TBD
+ * @package TEC\Events\Category_Colors
+ */
+
+namespace TEC\Events\Category_Colors;
+
+/**
+ * Helper class to manage meta keys for Category Colors.
+ */
+class Meta_Keys {
+
+	/**
+	 * Prefix for storing category meta values.
+	 *
+	 * @since TBD
+	 * @var string
+	 */
+	protected static string $meta_key_prefix = 'tec-events-cat-colors-';
+
+	/**
+	 * List of valid meta keys.
+	 *
+	 * @since TBD
+	 * @var array<string, string>
+	 */
+	protected static array $keys = [
+		'primary'          => 'primary',
+		'secondary'        => 'secondary',
+		'text'             => 'text',
+		'priority'         => 'priority',
+		'hide_from_legend' => 'hidden',
+	];
+
+	/**
+	 * Returns the full meta key with prefix.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key The key name (e.g., 'primary', 'background').
+	 *
+	 * @return string|null Full meta key or null if the key does not exist.
+	 */
+	public static function get_key( string $key ): ?string {
+		return self::$keys[ $key ] ?? null ? self::$meta_key_prefix . self::$keys[ $key ] : null;
```

Comment:
This line is a bit confusing to me, and took me a couple of times reading it through to understand it. I think this is a case where it would be better to be more verbose for the benefit of clarity.

I also think it would be better to avoid the mixed return type if possible, and throw an exception if the key isn't one that is recognized.